### PR TITLE
Barebone Alfresco subsystem

### PR DESCRIPTION
**Pull request by todorinskiz**
1. Added Alfresco Dependencies
2. Added Docker Build Image & Compose Up
3. Cleaned up obsolete code
4. Added an empty subsystem service
5. Improved threading and transnational resiliency
6. Added basic unit tests

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,37 +1,61 @@
...
+
     repositories {
         mavenCentral()
+        maven {
+            url "${rootProject.projectDir}/alfresco-bom-repo"
+        }
         maven {
             url 'https://artifacts.alfresco.com/nexus/content/groups/public'
         }
+        maven {
```
Comment (vierbergenlars) [inline build.gradle:53]: Do we need access to our internal artifactory?
Currently, I only see dependencies on public artifacts.

File: alfresco-health-processor-platform/build.gradle [hunk 2]
```diff
@@ -1,46 +1,30 @@
...
-}
+    compileOnly "org.projectlombok:lombok:${lombokVersion}"
+    annotationProcessor "org.projectlombok:lombok:${lombokVersion}"
 
-processResources {
-    filesMatching("**/module.properties") {
-        expand project.properties
-    }
-}
+    alfrescoProvided(enforcedPlatform("eu.xenit.alfresco:alfresco-community-bom:${alfrescoVersion}"))
```
Comment (tgeens) [inline alfresco-health-processor-platform/build.gradle:16]: it's not clear which `alfrescoVersion` is actually used here ?

File: alfresco-health-processor-platform/src/main/amp/config/alfresco/module/alfresco-health-processor/context/subsystem-context.xml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        <property name="interfaces">
+            <list>
+                <value>eu.xenit.alfresco.processor.service.ProcessorService</value>
+            </list>
+        </property>
+    </bean>
+    <bean id="healthProcessorPlatform.subsystem.type"
+          class="org.alfresco.repo.management.subsystems.ChildApplicationContextFactory"
+          parent="abstractPropertyBackedBean">
+    <property name="category" value="healthProcessor"/>
```
Comment (tgeens) [inline alfresco-health-processor-platform/src/main/amp/config/alfresco/module/alfresco-health-processor/context/subsystem-context.xml:19]: indents (can be fixed later)

File: alfresco-health-processor-platform/build.gradle (hunk 2 above)
Comment (todorinskiz) [reply alfresco-health-processor-platform/build.gradle:16]: Addressed!

File: alfresco-health-processor-platform/src/main/amp/config/alfresco/module/alfresco-health-processor/context/subsystem-context.xml (hunk 3 above)
Comment (todorinskiz) [reply alfresco-health-processor-platform/src/main/amp/config/alfresco/module/alfresco-health-processor/context/subsystem-context.xml]: Addressed!

File: build.gradle (hunk 1 above)
Comment (todorinskiz) [reply build.gradle]: Removed!

**Review by tgeens — APPROVED** [review]

**Pull request merged**